open-hackathon-server/src/hackathon/hack/host_server_manager.py

```diff
@@ -681,3 +681,38 @@ def __exist_request_host_server_by_hackathon_id(self, request_count, hackathon_i
                                        DockerHostServer.state == DockerHostServerStatus.DOCKER_READY,
                                        DockerHostServer.disabled == DockerHostServerDisable.ABLE)
         return len(vms) > 0
+
+
+class CloudServiceAvailable:
+    """
+    Since different docker hosts in the same cloud service can only create one container at the same time,
+    CloudServiceAvailable is used to check whether a docker host in a cloud service is creating a container.
+    """
+    __cloud_service_set = set()
+    __host_lock = thread.allocate_lock()
+
+    def isAvailable(self, public_ip):
+        """
+        check whether a docker host in a cloud service is ready to create a container.
+        :param public_ip: the public ip of a cloud service
+        :return: string such as 123.321.32.3
+        """
+        self.__host_lock.acquire()
```

Comment:
What's happened when a second thread try to acquire the lock and fails? Will it just wait until an lock got or it tries to pick another docker host server without waiting?